jobs/build/refresh-images/Jenkinsfile

```diff
@@ -94,42 +197,138 @@ node('openshift-build-1') {
-                 * By default, do not specify a version or release for oit. This will preserve the version label and remove
-                 * the release label. OSBS now chooses a viable release label to prevent conflicting with pre-existing
+                 * By default, do not specify a version or release for oit.
+                 * This will preserve the version label and remove
+                 * the release label. OSBS now chooses a viable release label
+                 * to prevent conflicting with pre-existing
                  * builds. Let's use that fact to our advantage.
                  */
-                buildlib.oit """
---working-dir ${OIT_WORKING} --group 'openshift-${OSE_MAJOR}.${OSE_MINOR}'
-images:update-dockerfile
-  ${oit_update_docker_args}
-  --message 'Updating for image refresh'
-  --push
-  """
-
-                buildlib.oit """
---working-dir ${OIT_WORKING} --group openshift-${OSE_MAJOR}.${OSE_MINOR}
-images:build
---push-to-defaults --repo-type signed
-"""
+                buildlib.oit(
+                    [
+                        "--working-dir ${OIT_WORKING}",
+                        "--group 'openshift-${OSE_MAJOR}.${OSE_MINOR}'",
+                        "images:update-dockerfile",
+                        "${oit_update_docker_args}",
+                        "--message 'Updating for image refresh'",
+                        "--push"
+                    ].join(' ')
+                )
+
+                // The image build can fail and still be acceptble.
+                // The waitUntil wrapper allows the user to respond and
+                // accept, retry or abort the build process and the task
+                BUILD_CONTINUED = false
+                waitUntil {
+                    try {
+
+                        // intially build everything.  Retrys will ignore
+                        // failed builds and only rebuild untried images
+                        exclude = ""
+                        if (BUILD_EXCLUSIONS != "") {
+                            exclude =
+                                "-x ${BUILD_EXCLUSIONS} --ignore-missing-base"
+                        }
+
+                        buildlib.oit(
+                            [
+                                "--working-dir ${OIT_WORKING}",
+                                "--group openshift-${OSE_MAJOR}.${OSE_MINOR}",
+                                exclude,
+                                "images:build",
+                                "--push-to-defaults --repo-type signed"
+                            ].join(' ')
+                        )
+                    }
+                    catch (err) {
+
+                        failed_builds = get_failed_builds(OIT_WORKING)
+
+                        mail(
+                            to: "${MAIL_LIST_FAILURE}",
+                            from: "aos-cicd@redhat.com",
+                            subject: "RESUMABLE Error during Refresh Images for OCP v${BUILD_VERSION}",
```

Comment:
BUILD_VERSION is being used, but not defined. This job works with OSE_MAJOR/OSE_MINOR which you can use to initialize a BUILD_VERSION variable. Alternatively, the job parameters can be reworked to match build/ocp (i.e. BUILD_VERSION list of 3.3, .... 3.10).